### ISSUE TITLE
Missleading status when replicating non-existing topic

### ISSUE THREAD

**Issue by twiechert** [Comcast/MirrorTool-for-Kafka-Connect#18]
Should show something like topic `XY` does not exist, ideally at connector startup.

Instead, spawning tasks fails silently, due to

```    
int taskCount = Math.min(maxTasks, leaderTopicPartitions.size());
        if (taskCount < 1) {
            LOG.warn("No tasks to start.");
            return new ArrayList<>();
}
```

Since leaderTopicPartitions => 0

**Comment by rhysmccaig** [comment]
Good pickup. Do you want to send a pull request to fix this? Otherwise I can get to it later in the week.

**Comment by twiechert** [comment]
I certainly can. I thereby propose to move the prerequisite (topics exists) checking to the connector startup. Otherwise, the connect connector status will always yield a RUNNING status.

Basically, the current prerequisites are then: Do all topics on source side exist, otherwise throw exception.

Would that be ok for you?

**Comment by rhysmccaig** [comment]
The problem with moving the check to connector startup is that the partition monitor thread will periodically poll the cluster for new topic partitions. 
One use case would be to set up a mirror task on a set of topics (or topic regex) that don't yet exist such as `projectname.*` where the topics may be created in the future. In this case it would not be desirable to stop the connector just because there are no available partitions.

I think there are 2 things we can do here:

1. (easy) Update the warning message to indicate that there are no topic partitions to mirror.
2. (a bit more work) Add an option to the connector to fail on startup if there are no topic partition available to mirror.

Thoughts?

**Comment by twiechert** [comment]
Yes, sounds reasonable. I totally ignored that the topic subscription is pattern based.
I would suggest option two is enough, essentially adding a warning if a provided pattern does not match any topic at task creation time.

**Comment by rhysmccaig** [comment]
Closed via #20
